### PR TITLE
docs: adds provisioning examples for all datasources

### LINKED ISSUE #11363: Documentation: add provisioning example to each data source page
Each core data source should have an example yaml snippet on how to provision it. This will make it easier for first time provisioning users to get started.

### PR DESCRIPTION
**Pull request by bergquist**
closes #11363

### CODE REVIEW

File: docs/sources/features/datasources/elasticsearch.md [hunk 1]
```diff
@@ -137,3 +137,22 @@ Query | You can leave the search query blank or specify a lucene query
 Time | The name of the time field, needs to be date field.
 Text | Event description field.
 Tags | Optional field name to use for event tags (can be an array or a CSV string).
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/elasticsearch.md]: typo

File: docs/sources/features/datasources/graphite.md [hunk 2]
```diff
@@ -120,3 +120,20 @@ queries via the Dashboard menu / Annotations view.
 
 Graphite supports two ways to query annotations. A regular metric query, for this you use the `Graphite query` textbox. A Graphite events query, use the `Graphite event tags` textbox,
 specify a tag or wildcard (leave empty should also work)
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/graphite.md]: typo

File: docs/sources/features/datasources/influxdb.md [hunk 3]
```diff
@@ -174,3 +174,21 @@ SELECT title, description from events WHERE $timeFilter order asc
 For InfluxDB you need to enter a query like in the above example. You need to have the ```where $timeFilter```
 part. If you only select one column you will not need to enter anything in the column mapping fields. The
 Tags field can be a comma separated string.
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/influxdb.md]: typo

File: docs/sources/features/datasources/mysql.md [hunk 4]
```diff
@@ -225,3 +225,20 @@ tags | Optional field name to use for event tags as a comma separated string.
 ## Alerting
 
 Time series queries should work in alerting conditions. Table formatted queries is not yet supported in alert rule conditions.
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/mysql.md]: typo

File: docs/sources/features/datasources/opentsdb.md [hunk 5]
```diff
@@ -88,3 +88,21 @@ Query | Description
 *tag_values(cpu, hostanme, env=$env, region=$region)* | Return tag values for cpu metric, selected env tag value, selected region tag value and tag key hostname
 
 For details on OpenTSDB metric queries checkout the official [OpenTSDB documentation](http://opentsdb.net/docs/build/html/index.html)
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/opentsdb.md]: typo

File: docs/sources/features/datasources/postgres.md [hunk 6]
```diff
@@ -217,3 +217,23 @@ tags | Optional field name to use for event tags as a comma separated string.
 
 Time series queries should work in alerting conditions. Table formatted queries is not yet supported in alert rule
 conditions.
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/postgres.md]: typo

File: docs/sources/features/datasources/postgres.md [hunk 7]
```diff
@@ -217,3 +217,23 @@ tags | Optional field name to use for event tags as a comma separated string.
...
+
+datasources:
+  - name: Postgres
+    type: postgres
+    url: localhost:5432
+    database: grafana
+    user: grafana
+    password: password
+    jsonData:
+      sslmode: "disable" # disable/require/verify-ca/verify-full
```
Comment (marefr) [inline docs/sources/features/datasources/postgres.md]: Nice!

File: docs/sources/features/datasources/prometheus.md [hunk 8]
```diff
@@ -100,3 +100,18 @@ The step option is useful to limit the number of events returned from your query
 ## Getting Grafana metrics into Prometheus
 
 Since 4.6.0 Grafana exposes metrics for Prometheus on the `/metrics` endpoint. We also bundle a dashboard within Grafana so you can get started viewing your metrics faster. You can import the bundled dashboard by going to the data source edit page and click the dashboard tab. There you can find a dashboard for Grafana and one for Prometheus. Import and start viewing all the metrics!
+
+## Configure datasource with provisioning
+
+Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources
+Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/prometheus.md]: typo

**Review by marefr — CHANGES_REQUESTED** [review]
Overall looks good. Some typos as pointed in comments. Think that link to *Configure datasource with provisioning* for all data sources from the provisioning page would be nice.

Some general thoughts that we should fix later/create issue for: We document all the settings you can do in UI for each data source, but not what their equivalent in JSON/yaml are.

**Comment by bergquist** [comment]
ping @marefr

File: docs/sources/features/datasources/elasticsearch.md [hunk 9]
```diff
@@ -140,7 +140,7 @@ Tags | Optional field name to use for event tags (can be an array or a CSV strin
 
 ## Configure datasource with provisioning
 
-Its now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
+It's now possible to configure datasources using config files with Grafanas [provisioning system](/administration/provisioning/#datasources).
 Here are some examples of how you can configure the Cloudwatch datasource using configuration.
```
Comment (marefr) [inline docs/sources/features/datasources/elasticsearch.md]: That fix was nice, didn't spot that :smile: 

But I was referring to the *Cloudwatch* reference. Must be changed for all data sources except Cloudwatch. Maybe we can skip the name of the data source and just use *configure this datasource for provisioning* or similar?

**Review by marefr — CHANGES_REQUESTED** [review]
You fixed another typo that I didn't spot :+1: But I was referring to another one. See comment on elasticsearch which are needed for all data source docs.

**Review by marefr — APPROVED** [review]

**Pull request merged**